udfs/tests/run.sh

```diff
@@ -15,23 +15,23 @@
 # limitations under the License.
 
 if [[ $1 == "--pip_install_before_run" ]]; then
```

Comment:
I recently had a look at this when adapting for current customer and noticed  a few issues:
1. not DRY (3x essentially the same code for no reason): solution [`shift`](https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_09_07.html)
1. only supports `-k` pytest option (should easily support all pytest options e.g. `-x` for fail fast): solution `$@` [Special Parameter](https://tiswww.case.edu/php/chet/bash/bashref.html#Special-Parameters)
1. The test scripts assumes `SHORT_SHA` env var is set (not the case when not in cloud build)
My suggestion is this (though you'd have to change references to SHORT_SHA to TEST_ID):

```bash
TEST_ID=$SHORT_SHA

# fall back on random string if SHORT_SHA not set
if [ -z "$TEST_ID" ]
then
  # shellcheck disable=SC2002
  TEST_ID=$(cat /dev/urandom | tr -dc 'a-z0-9_' | fold -w 32 | head -n 1)
fi

echo "Test ID: ${TEST_ID}"

if [[ $1 == "--pip_install_before_run" ]]; then
  shift
  python3 -m pip install -r tests/requirements.txt
fi
python3 tests/udf_test_utils.py --create_test_datasets
python3 -m pytest --workers 100 tests/create_udf_signatures.py $@
python3 -m pytest --workers 100 tests/test_create_udfs.py $@
python3 -m pytest --workers 100 tests/test_run_udfs.py $@
python3 tests/udf_test_utils.py --delete_test_datasets
```
Note that I might want to locally run the tests BEFORE committing so using short SHA of the most recent commit (git rev-parse --short HEAD) may be misleading.

Reply:
Great points and I didn't know about shift, thanks! 

With the build process now using a custom image and the testing steps being executed by the yaml file, the run.sh script is now just a helper script for the user to run tests locally. 

I've made the following updates to run.sh:
- Removed the pip install step since the contributing guide has users manually perform that step.
- Explicitly set the SHORT_SHA env variable in run.sh to a dummy value since it's only needed to create tests datasets in the users GCP project.